### PR TITLE
Add cat API for repositories and snapshots

### PR DESCRIPTION
**Pull request by ywelsch**
Adds the following two endpoints:
- /_cat/repositories
- /_cat/snapshots/{repository}

Relates to #13919

### CODE REVIEW

**Comment by dakrone** [comment]
This looks good, but can you add some rest tests for the two new _cat APIs?

**Comment by ywelsch** [comment]
Thanks for the review @dakrone. Can you have another look? I added tests and documentation.

**Comment by dakrone** [comment]
Left one really minor comment, other than that LGTM

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestSnapshotAction.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    @Override
+    protected Table getTableWithHeader(RestRequest request) {
+        return new Table()
+                .startHeaders()
+                .addCell("id", "alias:id,snapshotId;desc:unique snapshot id")
+                .addCell("status", "alias:s,status;text-align:right;desc:snapshot name")
+                .addCell("start_epoch", "alias:ste,startEpoch;desc:start time in seconds since 1970-01-01 00:00:00")
+                .addCell("start_time", "alias:sti,startTime;desc:start time in HH:MM:SS")
+                .addCell("end_epoch", "alias:ete,endEpoch;desc:end time in seconds since 1970-01-01 00:00:00")
+                .addCell("end_time", "alias:eti,endTime;desc:end time in HH:MM:SS")
```
Comment (drewr) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestSnapshotAction.java]: Can we also have a `duration` or a `elapsed` milliseconds? We can go ahead and do the math for people.

**Comment by gmile** [comment]
@ywelsch out of curiosity, how does this file gets generated? Is there a process to follow when adding new API endpoints?

https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/cat.repositories.json

**Pull request merged**